drivers/backup/portworx/portworx.go

```diff
@@ -68,6 +68,79 @@ type portworx struct {
 	token           string
 }
 
+var (
+	// AppParameters Here the len of "pre_action_list","pod_selector_list","background", "runInSinglePod", "container"
+	//should be same for any given app for a pre,post rule
+	AppParameters = map[string]map[string]map[string][]string{
+		"cassandra": {"pre": {"pre_action_list": {"nodetool flush -- keyspace1;", "echo 'test"},
+			"pod_selector_list": {"app=cassandra", "app=cassandra1"},
+			"background":        {"false", "false"},
+			"runInSinglePod":    {"false", "false"},
+			"container":         {"", ""},
+		},
+			"post": {"post_action_list": {"nodetool verify -- keyspace1;", "nodetool verify -- keyspace1;"},
+				"background":        {"false", "false"},
+				"pod_selector_list": {"app=cassandra", "app=cassandra1"},
+				"runInSinglePod":    {"false", "false"},
+				"container":         {"", ""},
+			},
+		},
+		"postgres": {"pre": {"pre_action_list": {"PGPASSWORD=$POSTGRES_PASSWORD; psql -U \"$POSTGRES_USER\" -c \"CHECKPOINT\";"},
+			"background":        {"false"},
+			"runInSinglePod":    {"false"},
+			"pod_selector_list": {"app=postgres"},
+			"container":         {"", ""},
+		},
+		},
+		"postgres-backup": {"pre": {"pre_action_list": {"PGPASSWORD=$POSTGRES_PASSWORD; psql -U \"$POSTGRES_USER\" -c \"CHECKPOINT\";"},
+			"background":        {"false"},
+			"runInSinglePod":    {"false"},
+			"pod_selector_list": {"app=postgres"},
+			"container":         {"", ""},
+		},
+		},
+		"mysql-backup": {"pre": {"pre_action_list": {"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH TABLES WITH READ LOCK;system ${WAIT_CMD};'"},
+			"background":        {"true"},
+			"runInSinglePod":    {"false"},
+			"pod_selector_list": {"app=mysql"},
+			"container":         {"", ""},
+		},
+			"post": {"post_action_list": {"mysql --user=root --password=$MYSQL_ROOT_PASSWORD -Bse 'FLUSH LOGS; UNLOCK TABLES;'"},
+				"background":        {"false"},
+				"pod_selector_list": {"app=mysql"},
+				"runInSinglePod":    {"false"},
```

Comment:
I thought you are creating single pre & post exec rules for all the applications.
Ideally it should be as below

Create single pre & post exec rule for all the applications
In testcase we just take decision whether you are going apply pre/post exec rule. Accordingly it will create a rule based on the application type. No need to pass the application list & rule name in the testcase.